src/__tests__/render.test.tsx

```diff
@@ -285,3 +285,17 @@ test('RenderAPI type', () => {
   render(<Banana />) as RenderAPI;
   expect(true).toBeTruthy();
 });
+
+test('should throw when rendering a string outside a text component', () => {
+  expect(() => render(<View>hello</View>)).toThrowError(
+    'Text strings must be rendered within a host Text component.'
+  );
+});
+
+test('should throw an error when rerendering with text outside of Text component', () => {
+  const { rerender } = render(<View />);
+
+  expect(() => rerender(<View>hello</View>)).toThrowError(
+    'Text strings must be rendered within a host Text component.'
+  );
+});
```

Comment:
We probably need to cover some other supported/not supported cases:
1. Support for components returning `string` value, e.g. i18next `Trans`, etc
```
function ReturnsString() { return 'Translated Text' };

function NestedReturnsString() { return <ReturnsStrings />: }

test("Should be fine", () => {
  render(<Text><ReturnsString /></Text>)
  render(<Text><NestedReturnsString /></Text>)
})

test("Should catch this", () => {
  expect(render(<View><ReturnsString /></View>)).toThrowError();
  expect(render(<View><NestedReturnsString /></View>)).toThrowError();
})
```
2. How does RN handle rendering `number` inside/outside of `<Text/>`. If it fails then we should match that behaviour.

Reply:
Rendering numbers also fails, I added a test case for it. Regarding the components returning string value you're right it will pass. I guess a solution would be a similar approach to the one I tried when trying to fix the same issue for queries meaning using the JSON representation in the assertStringsWithinText. This time there wouldn't be any change on the public api, the only issue is that there is no findAll method implemented for this type but it's not much of an issue

Reply:
If we ship this as experiemental then we can live with imperfect but still usable check. Seem like missing `toJSON` would solve all our problems

Reply:
I would keep the explicit fragment check until `toJSON` is avialable though

Reply:
In this case we wouldn't need the export of toJSON I think, we have access to it in the result of the render function as well as in the screen object so this should be doable

Reply:
I made the change, it now works with components returning strings and the implementation is way simpler

Reply:
Awesome that you managed to use `toJSON` 💯 , I thought that we would be limited by not being able to call `toJSON` on given node. It makes sense that we can do all the assertion on JSON tree instead of `ReactTestInstance` tree.